### ISSUE TITLE
chancloser: implement the new option_simple_close co-op close flow

### ISSUE THREAD

**Issue by Roasbeef** [lightningnetwork/lnd#8153]
We should implement the new restricted chan co-op close flow: https://github.com/lightning/bolts/pull/1096. Details of the new closing flow can be found in the linked PR. 

Once this is implemented, it should be using along side the _final_ taproot chans feature bit by default. This gives us a clean way to update to both the finalized version, and also require this new co-op close flow. 

Implementation wise, it may make sense to attempt to start from a clean slate (new state machine), so we can get a better feel of the total code surface area of this new variant vs the existing widely implemented co-op close flow.

**Comment by ZZiigguurraatt** [comment]
Seems like this has been superseded by https://github.com/lightning/bolts/pull/1205 and https://github.com/lightning/bolts/pull/1205 is now implemented in LND?

**Comment by Roasbeef** [comment]
Yeah you're right, we just never closed this issue after the feature was merged.
